.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,20 @@
+{
```

Comment:
any reason to add this file?

Reply:
Oh sorry, I should have explained. This was so I could develop the project on GitHub Codespaces (instead of using more disk space on my laptop). It's up to you if you want to merge it in or not, I can remove it from any future mergeable PRs. It's actually not foolproof because it doesn't install Stack by default yet. If you're interested I'll get it fully working, if not, I'll leave it as-is/remove it before we merge.

Reply:
On that topic, I'll want the opinion of @balacij .

Reply:
I'm not against this, but I'm hesitant to say we should merge until (at least):
1. We've cleaned up our own project configuration a little bit.
2. We can install all dependencies (for which we have many) through the file.
3. We write instructions on how to use it.
4. We confirm that it works for all of usual tasks and that we have a usecase. The [pricing of Codespaces](https://docs.github.com/en/billing/managing-billing-for-your-products/managing-billing-for-github-codespaces/about-billing-for-github-codespaces) doesn't look too bad, which is certainly appealing if we ever have students with low-spec computers!

Does HLS work with this? I would imagine that GitHub Codespaces has fairly limited resources?

Reply:
So we should definitely not merge this in via this particular PR.

Reply:
@balacij HLS works on Codespaces just fine, the smallest one has 2 vCores, 8gb of RAM and 32gb of storage. But I agree we shouldn't merge this yet.

Reply:
Also, there is a way to get "unlimited" (for practical purposes) Codespaces for free for education settings. It is nice for helping people get started for sure.